dspace-server-webapp/src/main/java/org/dspace/app/rest/repository/patch/operation/ResearcherProfileReplaceOrcidSyncPreferencesOperation.java

```diff
@@ -0,0 +1,156 @@
+
+import org.apache.commons.lang3.StringUtils;
+import org.dspace.app.orcid.service.OrcidSynchronizationService;
+import org.dspace.app.profile.OrcidEntitySyncPreference;
+import org.dspace.app.profile.OrcidProfileSyncPreference;
+import org.dspace.app.profile.OrcidSynchronizationMode;
+import org.dspace.app.profile.ResearcherProfile;
+import org.dspace.app.profile.service.ResearcherProfileService;
+import org.dspace.app.rest.exception.RESTAuthorizationException;
+import org.dspace.app.rest.exception.UnprocessableEntityException;
+import org.dspace.app.rest.model.patch.Operation;
+import org.dspace.authorize.AuthorizeException;
+import org.dspace.content.Item;
+import org.dspace.core.Context;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+/**
+ * Implementation for ResearcherProfile ORCID synchronization preferences
+ * patches.
+ *
+ * Example:
+ * <code> curl -X PATCH http://${dspace.server.url}/api/eperson/profiles/<:id-eperson> -H "
+ * Content-Type: application/json" -d '[{
+ *  "op": "replace",
+ *  "path": "/orcid/publications",
+ *  "value": "ALL"
+ *  }]'
+ * </code>
+ */
+@Component
+public class ResearcherProfileReplaceOrcidSyncPreferencesOperation extends PatchOperation<ResearcherProfile> {
+
+    private static final String OPERATION_ORCID_SYNCH = "/orcid";
+
+    private static final String PUBLICATIONS_PREFERENCES = "/publications";
+
+    private static final String FUNDINGS_PREFERENCES = "/fundings";
+
+    private static final String PROFILE_PREFERENCES = "/profile";
+
+    private static final String MODE_PREFERENCES = "/mode";
+
+    @Autowired
+    private ResearcherProfileService profileService;
+
+    @Autowired
+    private OrcidSynchronizationService synchronizationService;
+
+    @Override
+    public ResearcherProfile perform(Context context, ResearcherProfile profile, Operation operation)
+        throws SQLException {
+
+        String path = StringUtils.removeStart(operation.getPath(), OPERATION_ORCID_SYNCH);
+        String value = getNewValueFromOperation(operation);
+
+        Item profileItem = profile.getItem();
+
+        context.turnOffAuthorisationSystem();
+
+        try {
+            updatePreferences(context, path, value, profileItem);
+
+            return profileService.findById(context, profile.getId());
+
+        } catch (AuthorizeException e) {
+            throw new RESTAuthorizationException(e);
+        } finally {
+            context.restoreAuthSystemState();
+        }
+
+    }
+
+    private String getNewValueFromOperation(Operation operation) {
+        Object valueObject = operation.getValue();
+        if (valueObject == null | !(valueObject instanceof String)) {
+            throw new UnprocessableEntityException("The /orcid value must be a string");
+        }
+        return (String) valueObject;
+    }
+
+    private boolean updatePreferences(Context context, String path, String value, Item profileItem)
```

Comment:
As noted before, this method has an unclear return value, which means it should have JavaDocs.  What does "true" or "false" mean here?  Also, looking at the method usage, it doesn't look like it's return value is ever used, so this might just be updated to return void.